storagenode/piecestore/endpoint.go

```diff
@@ -361,7 +361,9 @@ func (endpoint *Endpoint) Upload(stream pb.DRPCPiecestore_UploadStream) (err err
 			mon.FloatVal("upload_failure_rate_bytes_per_sec").Observe(uploadRate)
 			if errors.Is(err, context.Canceled) {
 				// Context cancellation is common in normal operation, and shouldn't throw a full error.
-				endpoint.log.Info("upload failed", zap.Stringer("Piece ID", limit.PieceId), zap.Stringer("Satellite ID", limit.SatelliteId), zap.Stringer("Action", limit.Action), zap.Error(err), zap.Int64("Size", uploadSize), remoteAddrLogField)
+				endpoint.log.Info("upload canceled (context canceled)", zap.Stringer("Piece ID", limit.PieceId), zap.String("Reason", "Someone else uploaded the required piece faster."))
```

Comment:
It's also possible to get context.Canceled when the node is shutting down, so those log lines could be confusing when a bunch of them show up at the end of a log. And unfortunately it's not trivial to distinguish those cases. What do you think about this-

```go
endpoint.log.Info("upload canceled (race lost or node shutdown)", zap.Stringer("Piece ID", limit.PieceId))
```

Reply:
My bad there! I didn't find any documentation about that possibility, I've gone ahead and corrected it.